backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/UpdateVmCommand.java

```diff
@@ -953,15 +952,16 @@ private void updateVmNumaNodes() {
     }
 
     private void initNuma() {
+        if (VmCpuCountHelper.isResizeAndPinPolicy(getParameters().getVm())) {
+            getParameters().getVm().setvNumaNodeList(new ArrayList<>());
+            getParameters().setUpdateNuma(false);
```

Comment:
There is no point in setting an empty array list when you set update numa to false. It will just keep the old value. 

I think you should drop the following validation from `VmHandler::validateCpuPinningPolicy`
```
            if (vmFromDb != null
                    && (vmFromDb.getNumOfCpus() != vmFromParams.getNumOfCpus()
                            || !vmFromDb.getvNumaNodeList().equals(vmFromParams.getvNumaNodeList()))) {
                result = new ValidationResult(EngineMessage.ACTION_TYPE_CANNOT_RESIZE_AND_PIN_AND_NUMA_SET);
                break;
            }
```

And then to enforce the update by setting getParameters().setUpdateNuma(true);

Reply:
I do need the empty list, otherwise when the VM is running, I won't be able to edit the VM (same changes as I commented above). These are flows from the bug.

Reply:
I understand the need for the empty list, that is what this PR is about, but later in method `updateVmNumaNodes`  there is
```
    private void updateVmNumaNodes() {
        if (!getParameters().isUpdateNuma()) {
            return;
        }
```
so it seems to me that it will not be stored in the database.

Reply:
i switched to `getParameters().setUpdateNuma(true);`. in the validation i left only the CPU check, because if it comes from the API (changing CPU topology) we may will still need it. but if it still bothers it should be also safe to remove (since it's not UI, the NUMA won't change, anyway we override it and so on).

Reply:
done